packages/supi/src/install/index.ts

```diff
@@ -194,7 +194,8 @@ export async function mutateModules (
     let needsFullResolution = !maybeOpts.ignorePackageManifest && (
       !equals(ctx.wantedLockfile.overrides ?? {}, overrides ?? {}) ||
       !equals((ctx.wantedLockfile.neverBuiltDependencies ?? []).sort(), (neverBuiltDependencies ?? []).sort()) ||
-      ctx.wantedLockfile.packageExtensionsChecksum !== packageExtensionsChecksum)
+      ctx.wantedLockfile.packageExtensionsChecksum !== packageExtensionsChecksum) ||
+      opts.fixLockfile
```

Comment:
I was assuming this was the only change that had to be done.

Reply:
only change here do not fix requiresBuild field. because requiresBuild field always use pre existed lockfile for now. so I made a change, which preserve version but fix other fields. not just requiresBuild, but also other fields like hasBin and resolution(I also met the issue that the server's integrity  changed, and I can't get a good way to fix locked integrity. this can also fix this situation)

Reply:
if that is true, it would be nice to cover it with a test

Reply:
Adding test test the new --fix-lockfile? no problem, I will cover --fix-lockfile with a test, thanks